### PR TITLE
Storing data regarding incorrect input

### PR DESCRIPTION
**Pull request by sudheshna-donthineni**
- Every time a user types incorrect SRN, the user is displayed with a login error. But we did not have any information about whether the user continued to attempt a second time or even if they did, what the initial attempt was. 
- On the frontend, there are now three cases for `purpose_subtype`; `"liveclass"`, `"plio"`, `"incorrect-entry"`. 
- The only change in this code is adding another environment variable: `INCORRECT_ENTRY_TABLE_ID`, which stores the ID for the BQ table that this data is being fed to. 
- whenever the `purpose_subtype` is `"incorrect-login"`, we use that table ID.
- Another change is setting `is_multiple_entry`, if the length of userIDList > 1.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -5,8 +5,4 @@ repos:
     - id: flake8
       args: # arguments to configure flake8
         # making isort line length compatible with autopep8
-        - "--max-line-length=88"
--   repo: https://github.com/pre-commit/mirrors-autopep8
-    rev: 'v1.5.7'
-    hooks:
-    - id: autopep8
```
Comment (deepansh96) [inline .pre-commit-config.yaml:12]: why was this part removed?

File: lambda_function.py [hunk 2]
```diff
@@ -44,27 +44,29 @@ def lambda_handler(event, lambda_context):
...
                     row["purpose_type"] = message["purpose"]["type"]
-                    row["purpose_sub_type"] = message["purpose"]["subType"]
+                    row["purpose_subtype"] = message["purpose"]["subType"]
                     row["platform"] = message["purpose"]["params"]["platform"]
                     row["platform_id"] = message["purpose"]["params"]["id"]
                     row["auth_type"] = message["authType"]
                     row["user_id"] = each["userID"]
                     row["user_data_validated"] = each["valid"]
-
+                    print(row)
```
Comment (deepansh96) [inline lambda_function.py]: Is this for logging?

File: lambda_function.py [hunk 3]
```diff
@@ -44,27 +44,29 @@ def lambda_handler(event, lambda_context):
...
+                if len(message["user"]["values"]) > 1:
+                    row["is_multiple_entry"] = True
                 # parsing through each user ID and its respective valid flag
                 for each in message["user"]["values"]:
                     # the key values are the column names in the BQ table.
-                    row["timestamp"] = message["dateTime"]
+                    row["attendance_timestamp"] = message["dateTime"]
                     row["purpose_type"] = message["purpose"]["type"]
-                    row["purpose_sub_type"] = message["purpose"]["subType"]
+                    row["purpose_subtype"] = message["purpose"]["subType"]
```
Comment (deepansh96) [inline lambda_function.py:52]: This will change the column names in the BQ table right?
This might break the datastudio or the SQL queries right?

File: lambda_function.py [hunk 4]
```diff
@@ -44,27 +44,29 @@ def lambda_handler(event, lambda_context):
                 and (k in message["user"] for k in ("values"))
                 and (k in message["purpose"]["params"] for k in ("platform", "id"))
             ):
+                # sets is_multiple_entry
+                # if the length of user input list has more than 1 userID
+                if len(message["user"]["values"]) > 1:
+                    row["is_multiple_entry"] = True
```
Comment (deepansh96) [inline lambda_function.py]: Also, as pritam said, instead of this flag, we should have the number of entries

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: lambda_function.py (hunk 2 above)
Comment (sudheshna-donthineni) [reply lambda_function.py]: actually not really logging, just for testing. I removed it.

File: lambda_function.py (hunk 3 above)
Comment (sudheshna-donthineni) [reply lambda_function.py:52]: So, remember we created a new table of the attendance logs? I made the change in the lambda, but it wasn't updated in the repo. Hence, the change now.

File: lambda_function.py (hunk 4 above)
Comment (sudheshna-donthineni) [reply lambda_function.py]: Yeah, I'm not actually going to make this part of this PR now. Because I still need to test the whole staging flow. Will make a new PR later.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (sudheshna-donthineni) [reply .pre-commit-config.yaml:12]: I later learned that flake8 lets us know if there is an error. Whereas autopep8 automatically makes the changes. Wanted the former than the latter.

File: lambda_function.py (hunk 4 above)
Comment (deepansh96) [reply lambda_function.py]: Cool
But you'll make that PR before this flow is deployed on prod right?

File: lambda_function.py [hunk 5]
```diff
@@ -59,8 +56,8 @@ def lambda_handler(event, lambda_context):
                     row["auth_type"] = message["authType"]
                     row["user_id"] = each["userID"]
                     row["user_data_validated"] = each["valid"]
-                    print(row)
-                    return insert_data(row)
+
+                    insert_data(row)
```
Comment (deepansh96) [inline lambda_function.py:60]: Why is `return` removed? Does anything happen if we don't return from the main `lambda_handler` function?

File: lambda_function.py (hunk 4 above)
Comment (sudheshna-donthineni) [reply lambda_function.py]: https://github.com/avantifellows/auth-to-bigquery/pull/4

File: lambda_function.py (hunk 5 above)
Comment (sudheshna-donthineni) [reply lambda_function.py:60]: If you check the PR for multiple entries, I loop through the array. If I add a return it'll break the loop.
